src/ucp/core/ucp_ep.c

```diff
@@ -523,24 +527,32 @@ ucp_ep_create_api_conn_request(ucp_worker_h worker,
     }
 
     if (status == UCS_OK) {
-        *ep_p = ep;
         goto out;
     }
 
 out_ep_destroy:
     ucp_ep_destroy_internal(ep);
 out:
-    if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
-        if (status == UCS_OK) {
+    if (status == UCS_OK) {
+        if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
             status = uct_iface_accept(conn_request->uct.iface,
                                       conn_request->uct_req);
+        }
+    } else {
+        if (ucp_worker_sockaddr_is_cm_proto(worker)) {
+            uct_listener_reject(conn_request->uct.listener,
+                                conn_request->uct_req);
         } else {
             uct_iface_reject(conn_request->uct.iface, conn_request->uct_req);
         }
     }
 
     ucs_free(params->conn_request);
 
+    if (status == UCS_OK) {
+        *ep_p = ep;
```

Comment:
move this to line 547?

Reply:
Did you mean 541? then need to handle additionally status returned at line 538